.github/workflows/testing-and-deployment.yml

```
@@ -190,7 +198,15 @@ jobs:
         if: ${{ matrix.vtk-version == '9.1' || matrix.vtk-version == '9.0.3' }}
         run: pip install 'matplotlib<3.6'
 
+      - name: Set up numpy 2.0
+        if: ${{ matrix.numpy-version != 'default' }}
```

This may be nit-picky, feel free to ignore:

```suggestion
      - name: Install latest numpy 2.0 beta/rc
        if: ${{ matrix.numpy-version == 'latest' }}
```

---

Similarly for the other `continue-on-error` conditions, I'd make the choice of explicitly using `${{ matrix.numpy-version == 'latest' }}`

---

Mostly just thinking of a scenario where we specify the version of numpy being installed down the road